### PR TITLE
fixes OpenSSL error issue in ruby 1.9.3

### PR DESCRIPTION
**Pull request by dustinleblanc**
Closes https://github.com/Fullscreen/yt/issues/103

### CODE REVIEW

**Comment by bodrovis** [comment]
That's interesting because I was getting the same error with Ruby 2.

**Comment by dustinleblanc** [comment]
@bodrovis was the error intermittent? what version of 2.x were you using?

**Comment by bodrovis** [comment]
@dustinleblanc It was constant until I double-checked enabled services. I used Ruby 2.0.

File: lib/yt/request.rb [hunk 1]
```diff
@@ -200,14 +200,18 @@ def run_again?
...
         Errno::ENETUNREACH,
         Errno::ECONNRESET,
         Net::HTTPServerError
       ]
+      # OpenSSL::SSL::SSLErrorWaitReadable does not exist in Ruby 1.9.3
+      if defined?(OpenSSL::SSL::SSLErrorWaitReadable)
+        errors << OpenSSL::SSL::SSLErrorWaitReadable
+      end
+      errors
     end
```
Comment (claudiofullscreen) [inline lib/yt/request.rb:210]: @bodrovis I'm sorry for not accepting this PR before, I think this is the correct way.

However, do you mind changing the condition to `if RUBY_VERSION > '2'`? That's really the implicit requirement that we are making and that we are confirming in the tests. So I would like the method above to be:

``` ruby
def server_errors
  [
    OpenSSL::SSL::SSLError,
    Errno::ETIMEDOUT,
    Errno::ENETUNREACH,
    Errno::ECONNRESET,
    Net::HTTPServerError
  ] + (RUBY_VERSION < '2' ? [] : [OpenSSL::SSL::SSLErrorWaitReadable])
end
```

If you agree and change your commit to that, I will merge right away :dancers:

File: spec/models/request_spec.rb [hunk 2]
```diff
@@ -120,7 +120,7 @@
       # NOTE: This test is just a reflection of YouTube irrational behavior of
       # being unavailable once in a while, and therefore causing Net::HTTP to
       # fail, although retrying after some seconds works.
-      context 'an OpenSSL::SSL::SSLErrorWaitReadable' do
+      context 'an OpenSSL::SSL::SSLErrorWaitReadable', ruby2: true do
```
Comment (claudiofullscreen) [inline spec/models/request_spec.rb:123]: :heart_decoration: :heart_decoration: :heart_decoration:

File: lib/yt/request.rb (hunk 1 above)
Comment (bodrovis) [inline lib/yt/request.rb:210]: @claudiofullscreen Yeah that looks better however this is not my PR :)
@dustinleblanc

Comment (claudiofullscreen) [inline lib/yt/request.rb:210]: Oooops :joy_cat:

Comment (dustinleblanc) [inline lib/yt/request.rb:210]: Making change and re-running specs in 1.9 + 2.0 just to confirm nothing goes boom

File: lib/yt/request.rb [hunk 3]
```diff
@@ -239,10 +238,10 @@ def refresh_token_and_retry?
...
     def response_error
       case response
-        when *server_errors then Errors::ServerError
-        when Net::HTTPUnauthorized then Errors::Unauthorized
-        when Net::HTTPForbidden then Errors::Forbidden
-        else Errors::RequestError
+      when *server_errors then Errors::ServerError
+      when Net::HTTPUnauthorized then Errors::Unauthorized
+      when Net::HTTPForbidden then Errors::Forbidden
+      else Errors::RequestError
```
Comment (claudiofullscreen) [inline lib/yt/request.rb]: what happened here? Some automatic change added by your editor? :wink:

File: lib/yt/request.rb (hunk 1 above)
Comment (claudiofullscreen) [inline lib/yt/request.rb:210]: Thanks!

On Thu, Apr 9, 2015 at 9:40 AM, Dustin LeBlanc notifications@github.com
wrote:

> In lib/yt/request.rb
> https://github.com/Fullscreen/yt/pull/110#discussion_r28078830:
> 
> > ```
> >      Errno::ETIMEDOUT,
> >      Errno::ENETUNREACH,
> >      Errno::ECONNRESET,
> >      Net::HTTPServerError
> >    ]
> > ```
> > -      # OpenSSL::SSL::SSLErrorWaitReadable does not exist in Ruby 1.9.3
> > -      if defined?(OpenSSL::SSL::SSLErrorWaitReadable)
> > -        errors << OpenSSL::SSL::SSLErrorWaitReadable
> > -      end
> > -      errors
> >    end
> 
> Making change and re-running specs in 1.9 + 2.0 just to confirm nothing
> goes boom
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/Fullscreen/yt/pull/110/files#r28078830.

## 

 Claudio Baccigalupo  | Software Writer  |  Fullscreen, Inc.
http://Fullscreen.net  |  +1.310.202.3333

**Comment by claudiofullscreen** [comment]
And… can you squash your two commits into one? Thanks again!

File: lib/yt/request.rb (hunk 3 above)
Comment (dustinleblanc) [inline lib/yt/request.rb]: oh yeah... damn you `gg=G`!, guess I got one more fix ;)

**Comment by dustinleblanc** [comment]
NP, clearly my git-fu is getting rusty...too many `mend` commit messages are showing up lately :)

**Comment by dustinleblanc** [comment]
btw confirmed specs passed in 1.9.3 and 2.2.1

**Comment by claudiofullscreen** [comment]
:ribbon:

**Comment by dustinleblanc** [comment]
:+1:

**Pull request merged**